src/sync/installation.ts

```diff
@@ -45,7 +45,7 @@ export const getTargetTasks = (targetTasks?: TaskType[]): TaskType[] => {
 const getNextTask = async (subscription: Subscription, targetTasks?: TaskType[]): Promise<Task | undefined> => {
 	const tasks = getTargetTasks(targetTasks);
 
-	if (subscription.repositoryStatus !== "complete") {
+	if (subscription.repositoryStatus !== "complete" && tasks.includes("repository")) {
```

Comment:
/nit
Or alternatively, instead of modifying the allTaskTypes, we could just change this line?
`if (... && targetTasks.includes("repository")) {`

Reply:
I don't think this change is covered by a test...

Reply:
@gxueatlassian - allTaskTypes is the default list to backfill, so if we add the condition without adding it to allTaskTypes it will never do repo discovery unless we explicitly state, but we want it as default.

@bgvozdev - Adding test now :)